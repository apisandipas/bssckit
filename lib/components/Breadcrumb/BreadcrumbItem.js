"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _base = require("../Utilities/base");

var _themeFunctions = require("../../utils/themeFunctions");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  &:not(:first-child)::before {\n    display: inline-block;\n    padding-right: ", ";\n    padding-left: ", ";\n    color: ", ";\n    content: '/';\n  }\n  & + &:hover::before {\n    text-decoration: none;\n  }\n  ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    color: ", ";\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var active = function active(props) {
  return props.active && (0, _styledComponents.css)(_templateObject(), (0, _themeFunctions.getColor)(props, 'breadcrumbItem', 'color'));
};

var BreadcrumbItem = (0, _styledComponents["default"])(_base.Li)(_templateObject2(), function (props) {
  return (0, _themeFunctions.getPadding)(props, 'breadcrumbItem', 'right');
}, function (props) {
  return (0, _themeFunctions.getPadding)(props, 'breadcrumbItem', 'left');
}, function (props) {
  return (0, _themeFunctions.getColor)(props, 'breadcrumbItem', 'color');
}, active);
var _default = BreadcrumbItem;
exports["default"] = _default;