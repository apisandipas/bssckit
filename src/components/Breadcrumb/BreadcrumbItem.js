import styled, { css } from 'styled-components';
import { theme } from 'styled-tools';
import defaultTheme from '../../theme';
import { Li } from '../Utilities/base';

const getActive = props =>
  props.active &&
  css`
    color: ${theme('breadcrumbItem.colors.default.color')};
  `;

const BreadcrumbItem = styled(Li)`
  &:not(:first-child)::before {
    display: inline-block;
    padding-right: ${theme('breadcrumbItem.padding.right')};
    padding-left: ${theme('breadcrumbItem.padding.left')};
    color: ${theme('breadcrumbItem.colors.default.color')};
    content: '/';
  }
  & + &:hover::before {
    text-decoration: none;
  }
  ${getActive};
`;

BreadcrumbItem.defaultProps = {
  theme: defaultTheme,
};

export default BreadcrumbItem;
