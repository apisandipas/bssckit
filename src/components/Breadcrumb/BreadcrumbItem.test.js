import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import BreadcrumbItem from './BreadcrumbItem';

describe('Breadcrumb component', () => {
  it('matches BreadcrumbItem', () => {
    const { container } = render(<BreadcrumbItem />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('display', 'inline-block', {
      modifier: '&:not(:first-child)::before',
    });
    expect(container.firstChild).toHaveStyleRule('padding-right', '0.5rem', {
      modifier: '&:not(:first-child)::before',
    });
    expect(container.firstChild).toHaveStyleRule('padding-left', '0.5rem', {
      modifier: '&:not(:first-child)::before',
    });
    expect(container.firstChild).toHaveStyleRule('color', '#6c757d', {
      modifier: '&:not(:first-child)::before',
    });
    expect(container.firstChild).toHaveStyleRule('content', "'/'", {
      modifier: '&:not(:first-child)::before',
    });
  });

  it('matches BreadcrumbItem active', () => {
    const { container } = render(<BreadcrumbItem active />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('color', '#6c757d');
  });
});
