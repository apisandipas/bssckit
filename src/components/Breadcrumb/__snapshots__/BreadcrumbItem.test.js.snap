// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component matches BreadcrumbItem 1`] = `
.c1:not(:first-child)::before {
  display: inline-block;
  padding-right: 0.5rem;
  padding-left: 0.5rem;
  color: #6c757d;
  content: '/';
}

.c1 + .c0:hover::before {
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<li
  class="sc-AxhCb c0 c1"
/>
`;

exports[`Breadcrumb component matches BreadcrumbItem active 1`] = `
.c1 {
  color: #6c757d;
}

.c1:not(:first-child)::before {
  display: inline-block;
  padding-right: 0.5rem;
  padding-left: 0.5rem;
  color: #6c757d;
  content: '/';
}

.c1 + .c0:hover::before {
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<li
  class="sc-AxhCb c0 c1"
/>
`;
